plugin/adsk/plugin/geomNode.cpp

```diff
@@ -0,0 +1,236 @@
+    typedAttrFn.setArray(true);
+    typedAttrFn.setUsesArrayDataBuilder(true);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+    retValue = addAttribute(outGeomMatrixAttr);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+    //
+    // add attribute dependencies
+    //
+    retValue = attributeAffects(filePathAttr, outGeomAttr);
+    retValue = attributeAffects(rootPrimAttr, outGeomAttr);
+    retValue = attributeAffects(filePathAttr, outGeomMatrixAttr);
+    retValue = attributeAffects(rootPrimAttr, outGeomMatrixAttr);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+    return retValue;
+}
+
+MayaUsdGeomNode::MayaUsdGeomNode()
+    : MPxNode()
+    , _cache(std::make_unique<CacheData>())
+{
+    TfRegistryManager::GetInstance().SubscribeTo<MayaUsdGeomNode>();
+}
+
+/* virtual */
+MayaUsdGeomNode::~MayaUsdGeomNode() { }
+
+/* virtual */
+MStatus MayaUsdGeomNode::compute(const MPlug& plug, MDataBlock& dataBlock)
+{
+    MStatus retValue = MS::kSuccess;
+
+    if (plug == outGeomAttr || plug == outGeomMatrixAttr) {
+        MDataHandle filePathHandle = dataBlock.inputValue(filePathAttr, &retValue);
+        CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+        MDataHandle rootPrimHandle = dataBlock.inputValue(rootPrimAttr, &retValue);
+        CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+        const std::string fileName = TfStringTrim(filePathHandle.asString().asChar());
+        const std::string rootPrim = TfStringTrim(rootPrimHandle.asString().asChar());
+
+        if (!fileName.empty()) {
+            if (_cache->loadedFile.empty() || _cache->loadedFile != fileName
+                || _cache->rootPrim != rootPrim) {
+                MayaUsd::ImportData readData;
+                readData.setFilename(fileName);
+
+                if (!rootPrim.empty()) {
+                    // Set the root prim to import from
+                    readData.setRootPrimPath(rootPrim);
+                }
+
+                VtDictionary userArgs;
+                GfInterval   timeInterval;
+
+                UsdMayaJobImportArgs jobArgs = UsdMayaJobImportArgs::CreateFromDictionary(
+                    userArgs,
+                    /* importWithProxyShapes = */ false,
+                    timeInterval);
+
+                UsdMaya_MeshDataReadJob reader(readData, jobArgs);
+
+                std::vector<MDagPath> addedDagPaths;
+                reader.Read(&addedDagPaths);
+
+                _cache->primitives.swap(reader.mMeshData);
+                _cache->loadedFile = fileName;
+                _cache->rootPrim = rootPrim;
+            }
+
+            OutputArrayHandler geomOut(dataBlock, outGeomAttr, _cache->primitives.size(), retValue);
+            CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+            OutputArrayHandler matrixOut(
+                dataBlock, outGeomMatrixAttr, _cache->primitives.size(), retValue);
+            CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+            // MDataHandle outputGeomHandle = dataBlock.outputValue(outGeomAttr, &retValue);
```

Comment:
Code in comments voluntary?

Reply:
Oops, these are leftovers, I will remove them.